packages/engine/src/lib/services/variable-service.ts

```diff
@@ -127,6 +130,66 @@ export class VariableService {
     return Number(number);
   }
 
+  convertUrlOrBase64ToFile = async (urlOrBase64: unknown): Promise<ApFile | null> => {
+    if (isNil(urlOrBase64) || !isString(urlOrBase64)) {
+      return null;
+    }
+    // Get the file from the URL
+    try {
+
+
+      // Check if the string is a Base64 string
+      if (isBase64(urlOrBase64, { allowMime: true })) {
+        const matches = urlOrBase64.match(/^data:([A-Za-z-+/]+);base64,(.+)$/);
+        let base64 = urlOrBase64;
+        let contentType = null;
+
+        if (matches && matches?.length === 3) {
+          contentType = matches[1];
+          base64 = matches[2];
+
+          // You need to provide how you decide filename and extension in case of base64 string
+          const filename = 'unknown';
+          const extension = contentType.split('/')[1];
+
+          return {
+            filename: filename + "." + extension,
+            extension,
+            base64,
+          };
+        }
+
+      }
+      const response = await axios.head(urlOrBase64);
+      const contentType = response.headers['content-type'];
+
+      // Check if content type is file
+      if (!contentType || !(contentType.startsWith('application/') || contentType.startsWith("image") || contentType === 'application/octet-stream')) {
+        return null;
```

Comment:
quick question, why are file types restricted to these types? we can have for example `text/html` and many others, right?